### ISSUE TITLE
best settings / method to not get banned

### ISSUE THREAD

**Issue by snickers2k** [dweidenfeld/plexdrive#244]
Hey.
Thought PlexDrive is about not getting banned with GDrive. So i wonder why i still get banned.

> [Unit]
> Description=Mount Plexdrive to /mnt/Plexdrive
> After=syslog.target local-fs.target network.target
> [Service]
> Type=simple
> User=root
> ExecStartPre=-/bin/mkdir /mnt/Plexdrive
> ExecStart=/usr/local/bin/plexdrive mount \
>   -c /root/.plexdrive \
>   -o allow_other \
>   /mnt/Plexdrive
> ExecStop=/bin/fusermount -u -z /mnt/Plexdrive
> ExecStop=/bin/rmdir /mnt/Plexdrive
> Restart=always
> [Install]
> WantedBy=default.target

> [Unit]
> Description=Mount PlexDrive-Enrypted-GoogleDrive to /mnt/media
> After=syslog.target local-fs.target network.target
> [Service]
> Type=simple
> User=snikay
> #ExecStartPre=-/bin/mkdir /mnt/media
> ExecStart=/usr/bin/rclone mount \
>   --config /home/snikay/.config/rclone/rclone.conf \
>   --allow-other \
>   --allow-non-empty \
>   --buffer-size 1G \
>   --tpslimit 5 \
>   --tpslimit-burst 5 \
>   --fast-list \
>   --dir-cache-time 5000h \
>   --default-permissions \
>   --uid 1000 \
>   --gid 1000 \
>   --umask 0002 \
>   PlexDrive: /mnt/media
> ExecStop=/bin/fusermount -u /mnt/media
> #ExecStop=/bin/rmdir /mnt/media
> Restart=always
> [Install]
> WantedBy=multi-user.target

First i tried without the rclone settings, but got banned. so i thought reducing the amount of requests would work, but still getting banned.

Anyone having an idea? thanks.

**Comment by zmike808** [comment]
Do you have "Perform extensive media analysis during maintenance" on in plex's scheduled maintenance tasks? If so turn that off, and you should be fine.

**Comment by snickers2k** [comment]
ah, okay.
but isn't that negativ for plex's performance in some ways?

**Comment by zmike808** [comment]
Deep analysis is only used for certain specialized plex settings, the main one being for when you set "Limit remote stream bitrate". Plex does deep analysis to find out the exact bitrate at every moment in the file, instead of the default average bitrate for the entire file. If you're curious just google it and you can read up on exactly what it does. 

Regardless, it is currently impossible to use any gdrive mounting program with deep analysis on. You will always end up triggering a ban due to the nature of how deep analysis works. If you really want it just make a custom script to only run it on your local files.

TL;DR: no, it won't impact your performance.

**Comment by bFanek** [comment]
> Regardless, it is currently impossible to use any gdrive mounting program with deep analysis on. You will always end up triggering a ban due to the nature of how deep analysis works. 

False. I've had it turned on since plexdrive 2 and never triggered a ban.

**Comment by djsecrist** [comment]
I would double check to make sure you aren't using a regular rclone mount (I.e., not just to decrypt the PlexDrive mount) somewhere else. I had both mounted at first and forgot to switch my Radarr path over to the PlexDrive mount. When Radarr or Sonarr do their maintenance scans, it triggered a ban for me. I now use only PlexDrive and Unionfs with a script to automatically move everything from local to GD every 5 minutes. I haven't been banned in months!

**Comment by pcjonathan** [comment]
I would also echo the fact that I've got Deep Analysis turned on for PlexDrive (and rclone back when I used that) and never received any ban. Perhaps it's because I used fairly short maintenance windows.

**Comment by snickers2k** [comment]
Got banned daily, often after 20 minutes of plex's-scan... after turning off deep analysis, today is the first day i did not got banned .. Maybe it's because my connection is too fast? Have an 1GB/s connection , without limiting bandwidth for plexdrive/rclone

**Comment by dany20mh** [comment]
@snickers2k 
I’ve been using this since day one and I never get ban because of using this.

I do have 1Gbps internet speed which has nothing to do with this.

You either using something alongside of the PlexDrive which cause you ban or you do copy a lot from it.

**Comment by snickers2k** [comment]
With deep analysis turned on ?!
I do not have the Drive mounted with anything else than a single plexdrive mount.
In the Dashboard, there is clearly a high volume of requests when plex is performing scan. And after that, it got banned. although i'm still copying files from my home-server to gdrive, as well as new files from my root-server. But since i can still upload and view new files, it's only related to the Download-API (which google indeed confirmed). I'll try it now with deep-analysis turned off and report back. thanks for your help guys.

**Comment by djsecrist** [comment]
> although i'm still copying files from my home-server to gdrive, as well as new files from my root-server

How are you doing this?  This could actually be causing the ban, depending on what you are doing.  Remember that ANYTHING that is touching your Gdrive from ANYWHERE could be the culprit; it is not necessarily Plex.  Like I said, Sonarr was the offending app for me which was running on a completely different server than my Plex VPS.  I forgot about it at the time and was pulling my hair out thinking that Plex was causing the ban.

**Comment by snickers2k** [comment]
with "rclone move" with tps limiter to 1. 
but i'll try to do this seperate that.

**Comment by dany20mh** [comment]
@snickers2k 

I'm using rclone sync between my account, but you have to use bwlimit, cause tps is not effective and remember, Google has 10Tb Download and 750Gb Upload limit.

**Comment by snickers2k** [comment]
okaay... google refused to tell me the exact limits (!) :/ 
hmm.. thats strange. since my whole library is only 10TB right now (i'm still uploading), Plex can't reach that, right? or is it checking anything double and triple? :/
750GB Upload a is also not the case... hmpf.. bwlimit for what exactly? restricting the 10TB Download / 750GB Upload, or to restrict the requests per second?

thanks

**Comment by dany20mh** [comment]
@snickers2k 
Either you are new in this or you need to get more information about this.

Google has those 2 limit, and depend on what you use, to have to limit your usage, like if you do transfer file between account, you have to use --bwlimit=8.5M, which in your case that you uploading to your account is the same, after you pass that limit your account get ban and you can't upload to it anymore.

Same as download limit, if you use your account more than 10Tb, your account get lock and ban for 24 hours, it doesn't mean if you have a 8Tb account and you good to go and would never get ban, it's depend on your usage, rclone use it too much and download each single file over and over so never ever ever ever use rclone mount.

Hope these help you to understand the basics.

**Comment by snickers2k** [comment]
it does. thank you very much. i'll check my settings again.

**Comment by snickers2k** [comment]
oh, and since you don't use rclone-mount, do you use plexdrive for filebot as well?

**Comment by dany20mh** [comment]
I have PlexDrive, I use FileBot on it to rename my movies and tv show, I have Sonarr that is connected to it and I never had any ban issue other than if do try to upload too much or download too much or use other software alongside for mount.

**Comment by snickers2k** [comment]
okay. thank you.
and you have deep analysis turned on?

**Comment by dweidenfeld** [comment]
I can just speak for myself. I have everything turned on in plex and set the scan interval to 15 minutes. My library is ~20TB and I've never got banned right now.

Check if all of your programs with high load using plexdrive instead of a direct rclone mount ;)

**Comment by snickers2k** [comment]
hey. 
it indeed was the uploading to gdrive .. google said it was the "download-api" reaching the limit. but it was the upload.
thank you very much guys!
